README.md

```diff
@@ -8,6 +8,62 @@ a threshold.
 The ultimately use case for this script is to run it periodically and use it to
 trigger alerts/notifications that the Product Delivery team can take heed of.
 
+## Architectural Overview
+
+### In Plain English
+
+This is stateless script, written in Python, meant to be executed roughly once a day.
+It uses `requests.get()` to query [Redash](https://sql.telemetry.mozilla.org) for
+a query's data that is pre-emptively run every 24 hours. The data it analyses is a
+list of every collection stored in Remote Settings (Kinto) along with a count of
+every possible status (e.g. `up_to_date`, `network_error`, etc.)
+The script sums the "good" statuses and compares it against the "bad" statuses and
+if that ratio (expressed as a percentage) is over a certain threshold it alerts
+by sending an event to Sentry which notifies the team by email.
+
+### Historical Strategy
+
+As of Feb 2019, Firefox clients that use Remote Settings only send Telemetry events
+in daily batches. I.e. it's _not_ real-time. The "uptake histogram" is buffered in
+the browser and will send periodically instead of as soon as possible. In Firefox
+Nightly (67 at the time of writing), we are switching to real-time Telemetry Events.
+
+On the Telemetry backend we're still consuming the older uptake histogram but once,
+what is Nightly today, becomes Release, we will switch the Redash query (where
+appropriate) and still use this script to worry about the percentage thresholds.
+And the strategy for notifications should hopefully not change.
+
+Essentionally, once we switch to real-time Telemetry Events, nothing changes in
+terms of the surface API but the underlying data is more up-to-date and the response
+time of reacting to failure spikes is faster.
+
+### Redash
+
+The query we use is mentioned as a config default in the main `main.py` file.
+Follow that link, without the API key, in your browser to study the query and/or
+to fork it for a better query when applicable.
+
+### Configuration
+
+This script encodes all configuration inside `main.py` as defaults for things
+that can be overwritten by environment variables.
+
+The only **mandatory environment variable** is `REDASH_API_KEY`. You get this
+by logging in [https://sql.telemetry.mozilla.org](https://sql.telemetry.mozilla.org)
+and click to edit the/any relevant query and in the upper right-hand corner there's a
+drop-down with the label "..." and inside it "Show API Key".
```

Comment:
Maybe a screenshot would be relevant here?

Reply:
Meh. Too much work and hassle for something 2-3 people are ever going to hack on :)